genesis/sensors/base_sensor.py

```diff
@@ -202,6 +287,278 @@ def _get_formatted_data(
+
+
+@dataclass
+class NoisySensorMetadataBase(RigidSensorMetadataBase):
+    """
+    Utility base shared metadata class for analog sensors that are attached to a RigidEntity.
+    """
+
+    resolution: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    bias: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    bias_drift: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    bias_drift_std: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    noise_std: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    jitter_std_in_steps: torch.Tensor = field(
+        default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device)
+    )
+    jitter_in_steps: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    delay_in_steps: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    interpolate_for_delay: list[bool] = field(default_factory=list)
+
+
+class NoisySensorBase(RigidSensorBase):
+    """
+    Utility base sensor class for analog sensors that are attached to a RigidEntity.
+    """
+
+    @gs.assert_built
+    def set_bias(self, bias, envs_idx=None):
+        self._set_metadata_tensor(bias, self._shared_metadata.bias, envs_idx, self._cache_size)
+
+    @gs.assert_built
+    def set_bias_drift_std(self, bias_drift_std, envs_idx=None):
+        self._set_metadata_tensor(bias_drift_std, self._shared_metadata.bias_drift_std, envs_idx, self._cache_size)
+
+    @gs.assert_built
+    def set_noise_std(self, noise_std, envs_idx=None):
+        self._set_metadata_tensor(noise_std, self._shared_metadata.noise_std, envs_idx, self._cache_size)
+
+    @gs.assert_built
+    def set_jitter(self, jitter, envs_idx=None):
+        if isinstance(jitter, list):
+            jitter_in_steps = np.array(jitter) / self._manager._sim.dt
+        else:  # scalar or torch tensor or np array
+            jitter_in_steps = jitter / self._manager._sim.dt
+        self._set_metadata_tensor(jitter_in_steps, self._shared_metadata.jitter_std_in_steps, envs_idx, 1)
+
+    @gs.assert_built
+    def set_delay(self, delay, envs_idx=None):
+        self._set_metadata_tensor(delay, self._shared_metadata.delay_in_steps, envs_idx, 1)
+
+    def _set_metadata_tensor(self, input, metadata_tensor, envs_idx=None, n_per_sensor=1):
+        envs_idx = self._sanitize_envs_idx(envs_idx)
+        idx = self._sensor_idx * n_per_sensor
+        metadata_tensor[envs_idx, idx : idx + n_per_sensor] = self._sanitize_for_metadata_tensor(
+            input, shape=(len(envs_idx), n_per_sensor)
+        )
+
+    def _sanitize_for_metadata_tensor(self, input, shape) -> torch.Tensor:
+        if np.isscalar(input):
```

Comment:
Are you sure you need `np.isscalar`? As a reminder, `np.isscalar('numpy') = True`.

What about
```python
from numbers import Number

if isinstance(input, (Number, np.integer, np.floating, np.bool_)):
    [...]
```